src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -109,11 +109,15 @@ private void setS3EndpointType(String type) {
 
         if (isGlobal && S3_REGIONAL_ENDPOINT_VALUE.equals(s3EndpointSystemProp)) {
             System.clearProperty(S3_ENDPOINT_PROP_NAME);
-            refreshClientCache();
+            if (!clientCache.isEmpty()) {
+                refreshClientCache();
+            }
             logger.atDebug().log("s3 endpoint set to global");
         } else if (!isGlobal && !S3_REGIONAL_ENDPOINT_VALUE.equals(s3EndpointSystemProp)) {
             System.setProperty(S3_ENDPOINT_PROP_NAME, S3_REGIONAL_ENDPOINT_VALUE);
-            refreshClientCache();
+            if (!clientCache.isEmpty()) {
```

Comment:
this is not right.

Reply:
The problem is that you were trying to remove `null` from the map. The _region_ was null. So fix that problem in refreshClientCache.

Reply:
I realized, should I check the key region not null?